crates/core/component/ibc/src/component/rpc/connection_query.rs

```diff
@@ -96,12 +95,15 @@ impl<HI: HostInterface + Send + Sync + 'static> ConnectionQuery for IbcQuery<HI>
     }
 
     /// Connections queries all the IBC connections of a chain.
+    #[tracing::instrument(skip(self), err, level = "debug")]
     async fn connections(
         &self,
         _request: tonic::Request<QueryConnectionsRequest>,
     ) -> std::result::Result<tonic::Response<QueryConnectionsResponse>, tonic::Status> {
         let snapshot = self.storage.latest_snapshot();
-        let height = snapshot.version() + 1;
+        let height = HI::get_block_height(&snapshot)
```

Comment:
Is there an off-by-one here?

Reply:
I think so, but I am not sure what `ibc-go` does. They field it with https://github.com/cosmos/cosmos-sdk/blob/332d0b106d1b98538187b24394b912616c97b185/types/context.go#L73 so I think our behavior of doing `proof_height + 1` is coherent. I will ask to double check. Fixed, and thanks for the catch.

Reply:
Confirmed that's correct: https://github.com/cosmos/ibc-go/blob/main/modules/core/client/query.go#L68